From: Mike Turquette <mturquette@ti.com>
Date: Thu, 26 May 2011 22:41:17 +0000 (-0500)
Subject: CPUfreq: hotplug: offline aux CPUs below min freq
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=929e2018419786ee7ebd4892deb3ca88c9e681bb

CPUfreq: hotplug: offline aux CPUs below min freq

Previous offlining logic in cpufreq-hotplug would check if the current
frequency of the CPU matched its minimum frequency in the cpufreq_policy
structure.  In rare circumstances it is possible to run at a lower frequency
than the minimum and it is still desirable to take CPUs offline in these
circumstances.  Fixed by converting this checking from an exact match (==)
to a ranged match (<=).

Change-Id: Id3dfcf49be0fb3fe42e57f571827f7027fd6dc1d
Signed-off-by: Mike Turquette <mturquette@ti.com>
---

Index: kernel/drivers/cpufreq/cpufreq_hotplug.c
===================================================================
--- kernel.orig/drivers/cpufreq/cpufreq_hotplug.c	2011-05-30 17:18:28.209633213 +0900
+++ kernel/drivers/cpufreq/cpufreq_hotplug.c	2011-05-30 17:18:36.579599316 +0900
@@ -565,7 +565,7 @@
 	/* check for frequency decrease */
 	if (avg_load < dbs_tuners_ins.down_threshold) {
 		/* are we at the minimum frequency already? */
-		if (policy->cur == policy->min) {
+		if (policy->cur <= policy->min) {
 			/* should we disable auxillary CPUs? */
 #if 0
 			if (num_online_cpus() > 1 && hotplug_out_avg_load <
